crates/m3/src/gadgets/add.rs

```diff
@@ -142,6 +142,108 @@ impl U32Add {
 	}
 }
 
+/// A gadget for performing SIMD 32-bit integer addition on vertically-packed bit columns.
+///
+/// This gadget has input columns `xin` and `yin` for the two 32-bit adjacent integers to be added,
+/// and an output column `zout`, and it constrains that `xin + yin = zout` as integers. Only
+/// supports unchecked addition, so it does not handle overflow.
```

Comment:
Docs could be improved: Can you explain the user how to use V properly (at least limitations).